### PR TITLE
feature (auth/ldap): add Repository.CreateAuthMethod(...) and Repository.DeleteAuthMethod(...)

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

File: internal/auth/ldap/repository_auth_method_create.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+)
+
+func (r *Repository) CreateAuthMethod(ctx context.Context, am *AuthMethod, _ ...Option) (*AuthMethod, error) {
```
Comment (irenarindos) [inline internal/auth/ldap/repository_auth_method_create.go:13]: Can this be documented?

File: internal/auth/ldap/auth_method.go [hunk 2]
```diff
@@ -175,7 +183,7 @@ func (am *AuthMethod) convertUrls(ctx context.Context) ([]any, error) {
 		if err != nil {
 			return nil, errors.Wrap(ctx, err, op)
 		}
-		obj, err := NewUrl(ctx, am.PublicId, priority, parsed)
+		obj, err := NewUrl(ctx, am.PublicId, priority+1, parsed)
```
Comment (irenarindos) [inline internal/auth/ldap/auth_method.go:186]: Can you document that priority is incremented by 1 in this function/ why?

File: internal/auth/ldap/repository_auth_method_create.go (hunk 1 above)
Comment (jimlambrt) [reply internal/auth/ldap/repository_auth_method_create.go:13]: ty!

File: internal/auth/ldap/auth_method.go (hunk 2 above)
Comment (jimlambrt) [reply internal/auth/ldap/auth_method.go:186]: The `NewUrl(...)` documentation states that `connectionPriority cannot be less than one`.  That isn't sufficient, given that the range index will start at zero?

Comment (irenarindos) [reply internal/auth/ldap/auth_method.go:186]: That's fair! I was just thinking it'd be a little less digging for someone in the future

File: internal/auth/ldap/repository_auth_method_create.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	found, err := r.getAuthMethods(ctx, am.GetPublicId(), nil)
+	switch {
+	case err != nil:
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("failed to lookup created auth method %q", am.GetPublicId()))
+	case len(found) != 1:
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("found %q auth method (%q) and expected 1", len(found), am.GetPublicId()))
```
Comment (johanbrandhorst) [inline internal/auth/ldap/repository_auth_method_create.go:153]: `%q` is here used with an integer, but is only supported with strings and byte slices according to https://pkg.go.dev/fmt#hdr-Printing. I think you want `%d`?.

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]

File: internal/auth/ldap/repository_auth_method_create.go (hunk 3 above)
Comment (jimlambrt) [reply internal/auth/ldap/repository_auth_method_create.go]: ty!

**Review by irenarindos — APPROVED** [review]

**Review by johanbrandhorst — APPROVED** [review]

**Pull request merged**